app/com/linkedin/drelephant/tunin/AutoTuningAPIHelper.java

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.tunin;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import models.FlowDefinition;
+import models.FlowExecution;
+import models.JobDefinition;
+import models.JobExecution;
+import models.JobExecution.ExecutionState;
+import models.JobSuggestedParamValue;
+import models.TuningAlgorithm;
+import models.TuningJobDefinition;
+import models.TuningJobExecution;
+import models.TuningJobExecution.ParamSetStatus;
+import models.TuningParameter;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
+
+import play.libs.Json;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linkedin.drelephant.ElephantContext;
+import com.linkedin.drelephant.util.Utils;
+
+
+/**
+ * This class processes the API requests and returns param suggestion as response
+ */
+public class AutoTuningAPIHelper {
+
+  private static final Logger logger = Logger.getLogger(AutoTuningAPIHelper.class);
+
+  public static final String ALLOWED_MAX_RESOURCE_USAGE_PERCENT_DEFAULT =
+      "autotuning.default.allowed_max_resource_usage_percent";
+
+  public static final String ALLOWED_MAX_EXECUTION_TIME_PERCENT_DEFAULT =
+      "autotuning.default.allowed_max_execution_time_percent";
+
+  /**
+   * This class create a job execution with default parameters. This is required when there is no parameters set
+   * remains for suggestion.
+   * @param tuningJobDefinition
+   * @return
+   */
+  public TuningJobExecution createDefaultJobExecution(TuningJobDefinition tuningJobDefinition) {
+
+    //Get default execution from DB and clone that to create a new default execution
+    TuningJobExecution tuningJobExecutionDefault =
+        TuningJobExecution.find
+            .select("*")
+            .where()
+            .eq(TuningJobExecution.TABLE.jobExecution + "." + JobExecution.TABLE.job + "." + JobDefinition.TABLE.id,
+                tuningJobDefinition.job.id).eq(TuningJobExecution.TABLE.isDefaultExecution, true).setMaxRows(1)
+            .findUnique();
+
+    TuningJobExecution tuningJobExecution = new TuningJobExecution();
+    JobExecution jobExecution = new JobExecution();
+    jobExecution.id = 0L;
```

Comment:
Is 0 a dummy value? Does this get auto-incremented?

Reply:
this is for auto increment.